### ISSUE TITLE
IOVec.iov_len should be unsigned on all platforms

### ISSUE THREAD

**Issue by royneary** [nim-lang/Nim#15197]
When using ``IOVec`` from the ``posix`` module on some posix platforms (``linux+amd64``, ``nintendoswitch``), its ``iov_len`` field is declared as unsigned integer. On the other posix platforms (``macos+amd64``, ``openbsd+amd64`` and ``other``) it's declared as signed integer. So I can't compile the code below on all posix platforms.

### Example
```nim
from posix import IOVec
  
let buffer = newString(100)
let iov = IOVec(iov_base: buffer.cstring, iov_len: buffer.len.csize_t)
```

### Current Output
I am able to compile the code on ``linux+amd64`` but on ``linux+arm64`` I get
```
Error: type mismatch: got <csize_t> but expected 'int'
```

### Expected Output
The code should compile on all posix platforms.

### Possible Solution
All versions of ``IOVec`` should declare ``iov_len`` as unsigned integer. All C header files I found do so too (see e.g. [linux](https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git/tree/include/uapi/linux/uio.h?h=v5.8.1), [openbsd](https://github.com/openbsd/src/blob/master/sys/sys/uio.h), [macOS](https://github.com/apple/darwin-xnu/blob/master/bsd/sys/_types/_iovec_t.h)).

### Additional Information
```
$ nim -v
Nim Compiler Version 1.2.6 [Linux: amd64]
Compiled at 2020-07-29
Copyright (c) 2006-2020 by Andreas Rumpf

active boot switches: -d:release -d:nativeStackTrace
```

```
$ nim -v
Nim Compiler Version 1.2.6 [Linux: arm64]
Compiled at 2020-07-29
Copyright (c) 2006-2020 by Andreas Rumpf

active boot switches: -d:release -d:nativeStackTrace
```

**Comment by wltsmrz** [comment]
@royneary

**Comment by Clyybber** [comment]
Fixed in #15216
